src/Auth0.php

```diff
@@ -315,22 +315,28 @@ public function __construct(array $config)
             $this->idTokenAlg = $config['id_token_alg'];
         }
 
+        // Make sure the Client ID is included as a valid audience
+        $this->idTokenAud = [ $this->clientId ];
+
         // If a token audience is passed, make sure it's an array.
         if (! empty($config['id_token_aud'])) {
             if (! is_array( $config['id_token_aud'] )) {
                 throw new CoreException('Invalid id_token_aud; must be an array of string values');
             }
 
-            $this->idTokenAud = $config['id_token_aud'];
+            $this->idTokenAud = array_filter( array_merge( $this->idTokenAud, $config['id_token_aud'] ) );
         }
 
+        // Make sure the Domain is included as a valid issuer
+        $this->idTokenIss = [ 'https://'.$this->domain.'/' ];
+
         // If a token issuer is passed, make sure it's an array.
         if (! empty($config['id_token_iss'])) {
             if (! is_array( $config['id_token_iss'] )) {
                 throw new CoreException('Invalid id_token_iss; must be an array of string values');
             }
 
-            $this->idTokenIss = $config['id_token_iss'];
+            $this->idTokenIss = array_filter( array_merge( $this->idTokenIss, $config['id_token_iss'] ) );
```

Comment:
@joshcanhelp Interestingly the behavior I'm seeing is that the id token is always the auth0 url (while access token can be either depending which /authorize endpoint it's called; but that's out of scope for this PR).  I'm checking with protocol-sessions if that's the right behavior.

I don't think having an array of (trusted) issuers will be a problem as long as the array is not modified after the initial initialization and not blindly trusting the jwt claim. While unlikely, a client may expect 2 different tenants for authentication, and it might just be simpler to offer 1 sdk setup for 2 different tenants.

Reply:
If I use a custom domain, then that custom domain is used in the ID token:

**Custom domain (ROPG)**

![Screenshot 2019-08-29 15 49 38](https://user-images.githubusercontent.com/855223/63981616-b00a6880-ca74-11e9-8ee5-175ba9f9316e.png)

**Tenant domain (ROPG)**

![Screenshot 2019-08-29 15 51 06](https://user-images.githubusercontent.com/855223/63981690-ddefad00-ca74-11e9-8e93-ed0ddfca08ca.png)

Maybe I'm misunderstanding your comment?

Either way, I think we're in agreement that using the token issuer to pull the JWKS if fine if an invalid issuer will eventually fail the process. In other words, that was not a security risk in this case (though it's now pulling from the passed-in value).

Reply:
> using the token issuer to pull the JWKS if fine if an invalid issuer will eventually fail the process
No. **Don't trust the token contents until you've first verified the signature** and then the claims. And given that for the signature verification you'd need the key, you can't just take it from the `iss` claim of that still not verified token.

In the scenario of custom domains, normally the SDK needs to know not just the custom domain but also the auth0 one (In case the audience needs to be api v2 or userinfo, for example as those would keep that auth0 host). In that situation I think it should be OK to have an array of issuers. However.. if you know the token will always contain the same value in the `iss` claim, the custom domain value, then there's no need to allow a second one by using an array. 

The less doors we left open, the safer. I'm pinging @gkwang to confirm this via protocols-and-sessions, though you've already tested the results.

Reply:
> No. Don't trust the token contents until you've first verified the signature and then the claims.

As you can see in this PR, that's not happening. Just making the point that this part of the fix is not addressing a security issue, that's all.